### PR TITLE
wip: multiasset treasury pallet

### PR DESCRIPTION
**Pull request by tonyalaribe**
WIP: nothing to see yet

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2503542

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2503543

File: frame/treasury/src/types.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	Eq,
+	MaxEncodedLen,
+	TypeInfo,
+	Clone,
+	Copy,
+	Ord,
+	PartialOrd,
+	RuntimeDebug,
+)]
+pub enum NativeOrAssetBalance<AssetId, NativeBalance, AssetBalance> {
```
Comment (xlc) [inline frame/treasury/src/types.rs:61]: can we stop doing this? it shouldn't be the treasury's job to handle the difference between native and other assets. just build another pallet/helper for it.

Comment (tonyalaribe) [reply frame/treasury/src/types.rs:61]: I'm curious though. How would you approach it in a practical sense. Assuming it's a new pallet, how would it work?

Comment (xlc) [reply frame/treasury/src/types.rs:61]: https://github.com/paritytech/substrate/pull/12984#issuecomment-1362139263

File: frame/treasury/src/lib.rs [hunk 2]
```diff
@@ -453,9 +458,29 @@ pub mod pallet {
 			beneficiary: AccountIdLookupOf<T>,
 		) -> DispatchResult {
 			let max_amount = T::SpendOrigin::ensure_origin(origin)?;
-			let beneficiary = T::Lookup::lookup(beneficiary)?;
-
 			ensure!(amount <= max_amount, Error::<T, I>::InsufficientPermission);
+
```
Comment (tonyalaribe) [inline frame/treasury/src/lib.rs:462]: This change is carried over from a cherry picked commit and won't be in the final PR. Please ignore.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2512790

File: frame/treasury/src/types.rs (hunk 1 above)
Comment (tonyalaribe) [reply frame/treasury/src/types.rs]: Thanks! I've gone through it, and I'm exploring introducing a separate pallet for the multi asset functionality, which in turn actually makes things easier

**Comment by tonyalaribe** [comment]
With some exploration, it seems a simpler solution is making the treasury more generic to the payment approach. So, I'm closing this in favour of a different PR: https://github.com/paritytech/substrate/pull/13607/

**Pull request closed without merging**